node/rpc/src/state_api.rs

```diff
@@ -0,0 +1,411 @@
+// Copyright 2020 ChainSafe Systems
+// SPDX-License-Identifier: Apache-2.0, MIT
+
+use crate::RpcState;
+use actor::miner::{
+    compute_proving_period_deadline, ChainSectorInfo, DeadlineInfo, Deadlines, Fault, MinerInfo,
+    SectorOnChainInfo, SectorPreCommitOnChainInfo, State,
+};
+use address::Address;
+use async_std::task;
+use bitfield::json::BitFieldJson;
+use blocks::{tipset_json::TipsetJson, Tipset, TipsetKeys};
+use blockstore::BlockStore;
+use cid::{json::CidJson, Cid};
+use clock::ChainEpoch;
+use fil_types::SectorNumber;
+use jsonrpc_v2::{Data, Error as JsonRpcError, Params};
+use message::{
+    json::MessageReceiptJson,
+    unsigned_message::{json::UnsignedMessageJson, UnsignedMessage},
+};
+use serde::{Deserialize, Serialize};
+use state_manager::{InvocResult, MarketBalance, StateManager};
+use state_tree::StateTree;
+use wallet::KeyStore;
+
+#[derive(Serialize)]
+#[serde(rename_all = "PascalCase")]
+pub struct MessageLookup {
+    pub receipt: MessageReceiptJson,
+    pub tipset: TipsetJson,
+}
+
+#[derive(Serialize, Deserialize)]
+#[serde(rename_all = "PascalCase")]
+pub struct InvocResultJson {
+    pub msg: UnsignedMessageJson,
+    pub msg_rct: Option<MessageReceiptJson>,
+    pub actor_error: Option<String>,
```

Comment:
this should be renamed to error (right?)

Reply:
Good catch, I think the struct should have an error rather than an actor_error as per the spec